### ISSUE TITLE
Device available in adb, not in container

### ISSUE THREAD

**Issue by cnasc** [cezaryzelisko/flutter_dev#1]
Hi, big fan of this approach to remove differences between different dev machines. I'm running into an issue with device connectivity, however.

My phone is in developer mode, connected with "Use USB for File Transfer" in the settings, authenticated, and it shows up in `adb` on my host OS as follows:
```
$ adb devices
List of devices attached
D0AA003065JA0XXXXXX	device
```

However, inside VSCode when attached to the container, `flutter doctor -v` shows
```
[!] Connected device
    ! No devices available
```

[I modified the dockerfile](https://github.com/cnasc/flutter_dev/commit/c6c5f366e93d2296afa512f869d8f1cb146060ab) to include `build-essential` from `apt` and `ndk-bundle` from `sdkmanager`, which I need for my project. Any ideas on how to debug this/where I should start looking?

**Comment by cezaryzelisko** [comment]
Hi,
From time to time I have an connectivity issue (device isn't discovered in the container) when I connect my phone **before** the container is fully initialized. In that situation I simply detach my device and plug it in again when the container is up and running.

You can also try to run your app by directly selecting the desired device:
`flutter run -d <device_id>`
(This command should be run from your app's directory.)

**Comment by cnasc** [comment]
Hi, unfortunately neither of those options gave me any luck. I started from a fresh boot with the device unplugged, started the container in VSCode, then plugged in my device and it wasn't recognized. Then I tried running directly with the device ID and it wasn't able to find it.

I suspect there may be some other issues with my setup, but for the record in case anyone else comes along, I'm running Arch Linux, with `android-studio` from the AUR installed and `android-tools` (for `adb`) from the normal package archives.

I am at least able to build a debug apk from the container, which is more progress than I had running locally so thanks for providing an environment that unblocked me!

**Comment by cezaryzelisko** [comment]
Hi, I'm sorry to hear that you still can't connect your device.
You mentioned you're using Arch Linux, right? I've tested my solution on Ubuntu so maybe it's something with mounting?
In `.devcontainer/devcontainer.json` file there is a line which is responsible for mounting usb devices from host to the container ([link](https://github.com/cezaryzelisko/flutter_dev/blob/022c633aadb0b9d9e2659a2c7f422f9530c6330e/.devcontainer/devcontainer.json#L7)).
I'm not sure whether Arch Linux uses the same paths as Ubuntu.

**Comment by cnasc** [comment]
Just booted my machine to see if the mountpoints were different, and this time my device showed up! Not 100% sure if the order of operations matters here, but what I did (from a clean boot on my host) was:

1. Launch `adb` daemon (device not plugged in)
2. Open the workspace in VSCode
3. Connect my device
4. Accept prompt on device to allow debugging

At this point, `adb` on my host _did not_ see the device, but `flutter devices` in the container did. After that I was able to `flutter run` and it launched the app on my device.

So it looks like (on my machine at least) this is the chain of events that needs to happen for it to connect. Thanks so much for the help!
